src/crimson/os/seastore/cache.h

```diff
@@ -803,6 +826,7 @@ class Cache {
           if (!is_alive) {
             SUBDEBUGT(seastore_cache, "extent is not alive, remove extent -- {}", t, *e);
             remove_extent(e);
+	    e->state = CachedExtent::extent_state_t::INVALID;
```

Comment:
Looks it's very easy to miss on_invalidate().

How about introducing the following method?
```c++
void CachedExtent::set_invalid(Transaction &t) {
  state = CachedExtent::extent_state_t::INVALID;
  // I think it should be at the CachedExtent level to be symmetric with insertions to mutation_pendings
  if (trans_view_hook.is_linked()) {
      trans_view_hook.unlink();
  }
  on_invalidated(t);
}
```